patches/tModLoader/Terraria.ModLoader/ItemLoader.cs

```diff
@@ -261,11 +261,10 @@ internal static void DrawAnimatedItem(Item item, int whoAmI, Color color, Color
 			}
 		}
 
-		private static Rectangle AnimatedItemFrame(Item item)
+		internal static Rectangle ItemHoverHitbox(Item item)
 		{
-			int frameCount = Main.itemAnimations[item.type].FrameCount;
-			int frameDuration = Main.itemAnimations[item.type].TicksPerFrame;
-			return Main.itemAnimations[item.type].GetFrame(Main.itemTexture[item.type]);
+			Rectangle frame = Main.itemAnimations[item.type]?.GetFrame(Main.itemTexture[item.type]) ?? Main.itemTexture[item.type].Bounds;
+			return new Rectangle((int)(item.position.X + item.width * 0.5f - frame.Width * 0.5f), (int)(item.position.Y + item.height - frame.Height), frame.Width, frame.Height);
```

Comment:
Potential problem is if a modder decides to use different frame sizes. However, it is highly unlikely someone would ever do that. Therefore I let it pass.

Reply:
Yeah, most modders don't even make animated items, let alone override how they are drawn in the world. And if they really want different frame sizes, they could potentially create a custom class that inherits from DrawAnimation for it.